doc/extdev/nodes.rst

```diff
@@ -35,6 +35,34 @@ and in :py:class:`desc_signature_line` nodes.
 .. autoclass:: desc_optional
 .. autoclass:: desc_annotation
 
+Nodes for signature text elements
+.................................
+
+These nodes inherit :py:class:`desc_sig_element` and are generally translated
+to ``docutils.nodes.inline`` by :py:class:`!SigElementFallbackTransform`.
+
+When adding a new one, add it to :py:data:`SIG_ELEMENTS` via the class keyword
+argument `_sig_element=True` of :py:class:`desc_sig_element`, e.g.:
```

Comment:
We tell the user *what* to do here, but there isn't much detail as to *why* this should be done -- could you add a sentence please?